add support for custom llm with ssl support (#380)
katanemo/archgw: docs/source/concepts/llm_provider.rst

LLMs. Simply configure the details of the LLMs your application will use, and Arch offers a unified interface to
make outbound LLM calls.

Adding custom LLM Provider
--------------------------

> Out of the box we provide support for openai and mistral llm providers. But if you want to add your custom provider you can follow the below steps:

Reviewer: integrates with [any OpenAI-compatible API (e.g. Ollama)]

LLMs. Simply configure the details of the LLMs your application will use, and Arch offers a unified interface to
make outbound LLM calls.

Adding custom LLM Provider
--------------------------

We support any OpenAI compliant LLM for example mistral, openai, ollama etc. We offer first class support for openai and ollama. You can easily configure an LLM that communicates over the OpenAI API interface, by following the below guide. 

For example following code block shows you how to add an ollama-supported LLM in the `arch_config.yaml` file.
.. code-block:: yaml

    - name: local-llama
      provider_interface: openai
      model: llama3.2
      endpoint: host.docker.internal:11434

For example following code block shows you how to add mistral llm provider in the `arch_config.yaml` file.

.. code-block:: yaml

    - name: mistral-ai
      provider_interface: openai

>       model: llama3.2

Author: ```suggestion
      model: ministral-3b-latest
```

LLMs. Simply configure the details of the LLMs your application will use, and Arch offers a unified interface to
make outbound LLM calls.

Adding custom LLM Provider
--------------------------

> We support any OpenAI compliant LLM provider for example mistral, openai, ollama etc. Out of the box we provide support for openai and mistral. But if you want to add your custom provider you can follow the below steps:

Reviewer: ```suggestion
We support any OpenAI compliant LLM for example mistral, openai, ollama etc. We offer first class support for openai and ollama. You can easily configure an LLM that communicates over the OpenAI API interface, by following the below guide. 
```

> 1. Add the llm provider in the `llm_providers` section of the `arch_config.yaml` file. And supply details of how to reach to the provider using "endpoint" and specify `https` in `protocol`` if your endpoint is hosted on public cloud.

Reviewer: ```suggestion

```

> 

Reviewer: ```suggestion
For example following code block shows you how to add an ollama-supported LLM in the `arch_config.yaml` file.
```
